### PR TITLE
🏗 Update @percy/puppeteer to v2.x and fix incompatibilities in callers

### PR DESCRIPTION
**Pull request by danielrozenberg**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,11 +53,14 @@
     "@chiragrupani/karma-chromium-edge-launcher": "2.1.0",
     "@jest/core": "26.6.3",
     "@jridgewell/doctrine": "3.0.2",
+    "@percy/cli": "1.0.0-beta.39",
+    "@percy/puppeteer": "2.0.0",
```
Comment (rsimha) [inline package.json:57]: These two deps and the one below are large and should continue to live in the `visual-diff` directory. We shouldn't force every `amphtml` developer to install them. Curious about why they had to be moved out.

**Comment by danielrozenberg** [comment]
I haven't finished verifying this, but I'd like early feedback on this change.
The reason we took `puppeteer` out of the root `package.json` file was that it downloaded Chromium with it, and it wasn't necessary for all developers. Turns out we can just config it not to download by default, and handle the binary download in the visual diff code with less hacks around it :)

File: package.json (hunk 1 above)
Comment (rsimha) [reply package.json:57]: Saw your other comment. I still think visual-diff specific packages should not be moved to the root package. This is the model followed by every large testing tool / task.

Comment (danielrozenberg) [reply package.json:57]: See the comment below that took me to long to write 😅

Comment (danielrozenberg) [reply package.json:57]: For context:
```log
rodaniel@rodaniel99:~/amphtml (renovate/build-system-percy-puppeteer-2.x)$ rm -rf node_modules/
rodaniel@rodaniel99:~/amphtml (renovate/build-system-percy-puppeteer-2.x)$ npm ci

> amp-html@0.1.0 preinstall
> node build-system/common/check-package-manager.js

Detected node version v14.16.0 (latest LTS).
Detected gulp version 4.0.2.
Detected python version 3.8.7.
Detected npm version 7.6.1. Installing packages...

added 3647 packages in 17s
rodaniel@rodaniel99:~/amphtml (renovate/build-system-percy-puppeteer-2.x)$ du -h --summarize node_modules/puppeteer/
4.4M	node_modules/puppeteer/
rodaniel@rodaniel99:~/amphtml (renovate/build-system-percy-puppeteer-2.x)$ du -h --summarize node_modules/@percy/
3.8M	node_modules/@percy/
```

Comment (danielrozenberg) [reply package.json:57]: I strongly disagree, that model introduced an unnecessary level of complexity to our code - individual `gulp` tasks should not have to handle installing npm package...

Comment (estherkim) [reply package.json:57]: I like simpler code!! But I gotta agree with Raghu because the npm packages from all tasks will add up. Just off the top of my head, e2e has selenium-webdriver, chromedriver, geckodriver.

To be fair we should have probably added comments to the tasks to clarify.

Comment (rsimha) [reply package.json:57]: I see what you're getting at. Let me offer the full rationale:

- `amphtml` is a monorepo with a [dozen](https://github.com/ampproject/amphtml/search?q=%22package%22+in%3Apath&type=code) `package.json` files across sub-packages and sub-tasks.
- For deps that are part of a majority of developers' workflows, we've generally stuck them in the root `package.json`.
- While this has worked in the past, the directory size of `node_modules` is approaching 1 GB. There are two ways of dealing with this (that are actively being pursued):
    - Completely remove or replace outdated or unnecessary dependencies.
    - Move task-specific dependencies that aren't part of the default dev workflow out of the monorepo root.
- Every extra minute of work done during a CI job (e.g. installing packages) translates to half a million minutes of work (with a non-trivial $$ cost) across all CI jobs over a year. Moving deps needed by just one task / CI job will transfer the cost to all the other ~20 parallel jobs that don't need it.
- In the case of `tasks/visual-diff`, percy / puppeteer are task specific, and the entry in the root `.npmrc` is a detail of implementation that the root package installer should not have to worry about.
- Finally, for the correctness of CI checks, I think we should still have `puppeteer` download a known version of Chrome determined by its own version (or by a known override). This will help avoid non-deterministic results in future, say when a new version of Chrome with breaking changes is released.

**TL;DR:** My comment ended up longer and my push back stronger than I originally intended (sorry!), but I think there are enough good reasons to keep the current dependency model 😃

**Review by estherkim — APPROVED** [review]

File: build-system/tasks/visual-diff/index.js [hunk 2]
```diff
@@ -813,46 +823,8 @@ function installPercy_() {
...
-
-/**
- * @return {void}
- */
-function setupCleanup_() {
-  process.on('exit', cleanup_);
-  process.on('SIGINT', cleanup_);
-  process.on('uncaughtException', cleanup_);
-  process.on('unhandledRejection', cleanup_);
-}
```
Comment (rsimha) [inline build-system/tasks/visual-diff/index.js:827]: Nice to see this go away.

File: build-system/tasks/visual-diff/package.json [hunk 3]
```diff
@@ -4,11 +4,8 @@
...
   "description": "Gulp visual diff",
   "devDependencies": {
-    "@percy/agent": "0.28.6",
-    "@percy/puppeteer": "1.1.0",
+    "@percy/core": "1.0.0-beta.39",
+    "@percy/puppeteer": "2.0.0",
     "puppeteer": "8.0.0"
-  },
-  "resolutions": {
-    "**/**/minimist": "^1.2.3"
```
Comment (rsimha) [inline build-system/tasks/visual-diff/package.json:12]: Nice to see this go away too!

**Review by rsimha — APPROVED** [review]
Nice upgrade. LGTM!

File: package.json (hunk 1 above)
Comment (danielrozenberg) [reply package.json]: Alright, reverted that part of the change

**Comment by danielrozenberg** [comment]
There's one annoying little issue where `@percy/puppeteer` transitively downloads a different version of Chromium, before ending up re-using the one we ask it to use.

I spent some time trying to figure it out but couldn't find a clean way to handle it. I'll leave a TODO and open an issue with Percy

**Comment by danielrozenberg** [comment]
> There's one annoying little issue where `@percy/puppeteer` transitively downloads a different version of Chromium, before ending up re-using the one we ask it to use.
> 
> I spent some time trying to figure it out but couldn't find a clean way to handle it. I'll leave a TODO and open an issue with Percy

Never mind, I solved it :)

File: package.json [hunk 4]
```diff
@@ -57,6 +57,7 @@
     "@types/minimist": "1.2.1",
     "@types/mocha": "8.2.1",
     "@types/node": "14.14.32",
+    "0x": "4.10.2",
```
Comment (rsimha) [inline package.json:60]: @danielrozenberg Looks like this dependency was inadvertently added.

Comment (danielrozenberg) [reply package.json:60]: Merging package.json is always such a pain...

Comment (rsimha) [reply package.json:60]: I know! And this particular dep is weird because different versions of `npm` put it in different places. @alanorozco is being kind enough to take care of it in #33154.
